public/app/features/alerting/unified/state/AlertingQueryRunner.ts

```diff
@@ -81,15 +104,26 @@ export class AlertingQueryRunner {
   // to do this we will convert the list of queries into a DAG and walk the invalid node's output edges recursively
   async prepareQueries(queries: AlertQuery[]) {
```

Comment:
The logic behind this code seems fairly complex. Could we add a few tests covering that?

Reply:
Test included!